### PR TITLE
Improve documentation, fix error messages

### PR DESCRIPTION
**Pull request by abhikeshav**
* Improve documentation. Add read+delete examples.
 * Fix error messages to make it more clear
 * Fix cmake flags to remove clang hardcoding & coverage

### CODE REVIEW

File: sdk/cpp/samples/CMakeLists.txt [hunk 1]
```diff
@@ -19,12 +19,12 @@ set(samples bgp_create
                 bgp_xr_read
                 bgp_xr_opendaylight
                 bgp_restconf
-                xe_native_read)
```
Comment (ylil93) [inline sdk/cpp/samples/CMakeLists.txt]: Was this intentional?

File: sdk/python/core/docsgen/faq.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+The `release notes <https://github.com/CiscoDevNet/ydk-py/releases>`_ contains some information about which bundles were changed in which release of YDK.
+
+Also, the pypi page provides some information about this under the “Requires Distributions” section. For example, see:
+
+ - https://pypi.python.org/pypi/ydk-models-cisco-ios-xr/6.3.1
+ - https://pypi.python.org/pypi/ydk-models-cisco-ios-xr/6.2.2
+
+Also, we use the `ydk-gen <https://github.com/CiscoDevNet/ydk-gen>`_ tool to generate the bundles. This tool is available for anyone to use in order to generate the bundle version in combination with ydk core version of their choice. For example, the below steps will generate & install the ``cisco-ios-xr 6.3.1`` bundle compatible with ``ydk core 0.6.2`` (assuming you have already installed the `system dependencies <https://github.com/CiscoDevNet/ydk-py#system-requirements>`_):
+
+1) Install libydk (taking CentOS as example)
```
Comment (ylil93) [inline sdk/python/core/docsgen/faq.rst]: FAQ looks great. This section could be improved if you linked to the README containing instructions to install ydk on all the systems other than CentOS or similar documentation.

https://github.com/CiscoDevNet/ydk-gen/tree/master/sdk/cpp#quick-install

File: sdk/cpp/core/src/codec_service.cpp [hunk 3]
```diff
@@ -66,22 +64,15 @@ CodecService::encode(CodecServiceProvider & provider, Entity & entity, bool pret
         XmlSubtreeCodec codec{};
         return codec.encode(entity, root_schema);
     }
-    try
```
Comment (ylil93) [inline sdk/cpp/core/src/codec_service.cpp]: Is there a reason for eliminating the try catch around this code?

File: sdk/cpp/samples/CMakeLists.txt (hunk 1 above)
Comment (ylil93) [reply sdk/cpp/samples/CMakeLists.txt]: comment out instead

File: sdk/cpp/core/src/codec_service.cpp (hunk 3 above)
Comment (abhikeshav) [reply sdk/cpp/core/src/codec_service.cpp]: This hid the real issue from the user. User can have try-catch in their app

File: sdk/cpp/samples/CMakeLists.txt (hunk 1 above)
Comment (abhikeshav) [reply sdk/cpp/samples/CMakeLists.txt]: Done

**Comment by abhikeshav** [comment]
Thanks for your review @ylil93 !

**Pull request merged**